templates/kubernetes/terraform/global/remote-state/main.tf

```diff
@@ -0,0 +1,24 @@
+provider "aws" {
```

Comment:
To create the actual backend bucket and dynamo db. This will need to be run in a separate `tf apply` before the `tf init` of the rest of the infra. This will be an interesting piece to automate..

Reply:
We can move it to a seperate `.tf` file and use [terraform apply target](https://www.terraform.io/docs/commands/apply.html#target-resource) to apply specifically only that portion

Reply:
That wouldn't be a great workflow going forward though, because you wouldn't be able to run plan/apply in the main dir.

Reply:
How about using the `depends on` clause?

Reply:
They have to be run separately, because one creates the global state and should never be run again, and the other one uses the global state.

Reply:
Right, does it make sense to enclose the backend init in a shell script then or boto3?

Reply:
That way we can both spin up and tear down the backend init based on `tf apply` and `tf destroy`

Reply:
Gonna stick with this for now and we'll see how complex the process gets to see if it warrants adding helpers..